aclk/aclk_query.c

```diff
@@ -0,0 +1,574 @@
+#include "aclk_common.h"
+#include "aclk_query.h"
+#include "aclk_stats.h"
+
+pthread_cond_t query_cond_wait = PTHREAD_COND_INITIALIZER;
+pthread_mutex_t query_lock_wait = PTHREAD_MUTEX_INITIALIZER;
+#define QUERY_THREAD_LOCK pthread_mutex_lock(&query_lock_wait)
```

Comment:
```suggestion
#define QUERY_THREAD_LOCK(X) pthread_mutex_lock(&query_lock_wait)
```

Reply:
?

Reply:
It's just a suggestion. With this definition you can use the macro like a function in the code:
```
QUERY_THREAD_LOCK();
```
This reads like a function that does stuff.

Reply:
Oh OK I see you mean code readability. Well, I followed the pattern in ACLK that already existed which uses it without brackets. If we want to change it we have to do it for all the locks then in ACLK.

Reply:
I have not requested changes :)